tests/test_shuffle.py

```diff
@@ -1,12 +1,70 @@
+import uuid
+
 import dask
+import dask.dataframe as dd
+import pytest
+
+
+@pytest.fixture(scope="session")
+def s3_stability_url(s3, s3_bucket):
+    # Ensure that the stability-scratch directory exists,
+    # but do NOT reomove it as multiple test runs could be
+    # accessing it at the same time
+    stability_url = f"{s3_bucket}/stability-scratch"
+    s3.mkdirs(stability_url, exist_ok=True)
+    return stability_url
```

Comment:
Nice!

Is there a reason we this top-level `stability-scratch/` directory? If not, my preference would be to have each test session create its own unique bucket and then have each test create its own bucket inside the session bucket that it uses (similar to what's outlined in https://github.com/coiled/coiled-runtime/pull/67#issuecomment-1101567089). 

These s3 fixtures will also be used by other tests outside of this module. Let's move them over to `conftest.py`, and remove any stability references, so they can be used throughout the test suite, similar to our existing client fixture.

Reply:
> If not, my preference would be to have each test session create its own unique bucket

Do you mean a unique directory within the `dask-io` bucket or specifically a unique bucket `dask-test-1234abcd`? If the latter, I'm not sure if the AWS credentials have the requisite permissions and this would involve the use of boto3. 

If the former:

- Each test does have it's own unique directory, within the now renamed  top-level `test-scratch` directory
- I've created the `test-scratch` dir because Naty [pointed out](https://github.com/ncclementi) that if the GHA is cancelled lots of test data cruft can be left lying around.
- I wanted to avoid potential cruft in the bucket top-level directory structure.
- I wanted to signal that `test-scratch` can be deleted.

> These s3 fixtures will also be used by other tests outside of this module. Let's move them over to conftest.py, and remove any stability references, so they can be used throughout the test suite, similar to our existing client fixture.

I've done this and will discuss this in related comments.